pkg/cmd/cli/cmd/observe/observe.go

```diff
@@ -0,0 +1,936 @@
+	--each command as arguments (all custom arguments are omitted).
+
+More complicated interactions build on the two examples above - your inventory
+script could make a call to allocate storage on your infrastructure as a
+service, or register node names in DNS, or set complex firewalls. The more
+complex your integration, the more important it is to record enough data in the
+remote system that you can identify when resources on either side are deleted.
+
+Experimental: This command is under active development and may change without notice.`
+
+	observeExample = `	# observe changes to services
+	%[1]s observe services
+
+	# observe changes to services, including the clusterIP and invoke a script for each
+	%[1]s observe services -a '{ .spec.clusterIP }' --each register_dns.sh
+`
+)
+
+// NewCmdObserve creates the observe command.
+func NewCmdObserve(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ObserveOptions{
+		baseCommandName: fullName,
+		retryCount:      2,
+		templateType:    "jsonpath",
+		maximumErrors:   20,
+		listenAddr:      ":11251",
+	}
+
+	cmd := &cobra.Command{
+		Use:     "observe",
+		Short:   "Observe changes to resources and invoke integrations",
+		Long:    fmt.Sprintf(observeLong, fullName),
+		Example: fmt.Sprintf(observeExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(f, args, out, errOut); err != nil {
+				cmdutil.CheckErr(err)
+			}
+
+			if err := options.Validate(args); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(cmd, err.Error()))
+			}
+
+			if err := options.Run(); err != nil {
+				cmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	// flags controlling what to select
+	cmd.Flags().BoolVar(&options.allNamespaces, "all-namespaces", false, "If present, list the requested object(s) across all projects. Project in current context is ignored.")
+
+	// the commands to execute on changes
+	cmd.Flags().VarP(&options.eachCommand, "each", "e", "A command to run for each change. Specify mulitple times to add arguments.")
+
+	// to perform deletion observehronization
+	cmd.Flags().Var(&options.nameSyncCommand, "names", "A command that will list all of the currently known names, optional. Specify multiple times to add arguments. Use to get notifications when objects are deleted.")
+	cmd.Flags().BoolVar(&options.includeDeletes, "include-deletes", false, "If true, the --each command will be called for deleted objects. Some deleted objects will not have generated arguments beyond namespace and name.")
+
+	// add additional arguments / info to the server
+	cmd.Flags().StringVar(&options.templateType, "use", options.templateType, "Format for template arguments. jsonpath and gotemplate are supported.")
+	cmd.Flags().BoolVar(&options.strictTemplates, "strict-templates", false, "If true return an error on any field or map key that is not missing in a template.")
```

Comment:
If you are only selecting fields, then this should never fail.

On Tue, Aug 30, 2016 at 2:44 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/observe/observe.go
> <https://github.com/openshift/origin/pull/4196#discussion_r76855438>:
>
> > +		},
> > +	}
> > +
> > +	// flags controlling what to select
> > +	cmd.Flags().BoolVar(&options.allNamespaces, "all-namespaces", false, "If present, list the requested object(s) across all projects. Project in current context is ignored.")
> > +
> > +	// the commands to execute on changes
> > +	cmd.Flags().VarP(&options.eachCommand, "each", "e", "A command to run for each change. Specify mulitple times to add arguments.")
> > +
> > +	// to perform deletion observehronization
> > +	cmd.Flags().Var(&options.nameSyncCommand, "names", "A command that will list all of the currently known names, optional. Specify multiple times to add arguments. Use to get notifications when objects are deleted.")
> > +	cmd.Flags().BoolVar(&options.includeDeletes, "include-deletes", false, "If true, the --each command will be called for deleted objects. Some deleted objects will not have generated arguments beyond namespace and name.")
> > +
> > +	// add additional arguments / info to the server
> > +	cmd.Flags().StringVar(&options.templateType, "use", options.templateType, "Format for template arguments. jsonpath and gotemplate are supported.")
> > +	cmd.Flags().BoolVar(&options.strictTemplates, "strict-templates", false, "If true return an error on any field or map key that is not missing in a template.")
>
> Why would someone choose this? It feels like a "make it fail" flag. I
> could see setting an env var when you call the --each though. Seems like
> that would be best of both worlds.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/4196/files/69ab95173b3a274314ae59c42690a33690f9f69e#r76855438>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p-rIpSHNjY3SAhum-ilH62eoz27Yks5qlHomgaJpZM4Fsaph>
> .
>